php_phongo.c

```diff
@@ -1107,12 +1188,30 @@ static bool php_phongo_apply_rp_options_to_uri(mongoc_uri_t *uri, bson_t *option
 
 	new_rp = mongoc_read_prefs_copy(old_rp);
 
-	if (bson_iter_init_find_case(&iter, options, MONGOC_URI_SLAVEOK) && BSON_ITER_HOLDS_BOOL(&iter) && bson_iter_bool(&iter)) {
-		mongoc_read_prefs_set_mode(new_rp, MONGOC_READ_SECONDARY_PREFERRED);
+	if (bson_iter_init_find_case(&iter, options, MONGOC_URI_SLAVEOK)) {
+		if (!BSON_ITER_HOLDS_BOOL(&iter)) {
+			PHONGO_URI_INVALID_TYPE(iter, "boolean");
```

Comment:
Note: unlike generic boolean options (e.g. "ssl"), booleans for read preferences and write concerns were previously ignored if an unexpected type. I don't consider it a BC error to start throwing an error here, since libmongoc does the same during URI string parsing. See also: "journal" and "safe" for write concerns.